scripts/modowners/modowners_generation_script_test.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"testing"
+)
+
+/*
+question: how do i mock files that import the below 3 imports and use said imports?
+*/
+func TestGetFiles(t *testing.T) {
+	for _, test := range []struct {
+		moduleName     string
+		expectedResult []string
+	}{
+		{"cloud.google.com/go/storage v1.28.1", []string{"file1.go", "file2.go", "file3.go"}},
+		{"cuelang.org/go v0.5.0", []string{"file4.go"}},
+		{"github.com/Azure/azure-sdk-for-go v65.0.0+incompatible", []string{"file2.go", "file4.go", "file5.go"}},
+	} {
+		result, err := getFiles(test.moduleName)
```

Comment:
if you want to actually test this you probably need to pass in a root path as a parameter, that way you can create a 'test' directory structure and file's with fake modowners to test against.

Reply:
noted - will implement in the future when i get there